### PR TITLE
improve derive for registers (breaking)

### LINKED ISSUE #73: derive_register is copy
https://github.com/stm32-rs/svdtools/blob/6a8717f01c638a98e1abac0dbb10aaf08c4aae68/svdtools/patch.py#L781

As I can see `derive_register` and `derive_peripheral` are semantically different. It is more similar to `copy_peripheral`

### PR DESCRIPTION
**Pull request by burrbull**
Closes #73, #116

### CODE REVIEW

**Comment by burrbull** [comment]
@mciantyre Could you test both Python and Rust `svdtools`?

**Comment by burrbull** [comment]
svd2rust now supports register deriving, but svdtools doesn't
@adamgreig what your thoughts?

File: src/patch/peripheral.rs [hunk 1]
```diff
@@ -343,6 +357,47 @@ impl PeripheralExt for Peripheral {
...
+            (
+                rderive,
+                make_register(hash)?.derived_from(Some(rderive.into())),
+            )
+        } else {
+            return Err(anyhow!("derive: incorrect syntax for {}", rname));
+        };
+        self.get_register(rderive)
+            .ok_or_else(|| anyhow!("register {} not found", rderive))?;
+
```
Comment (burrbull) [inline src/patch/peripheral.rs:382]: Also 1 question. Should we do this check?
It doesn't support cross-peripheral deriving.

**Comment by mciantyre** [comment]
Looks good for my purposes. See 0faef6f744ba692c4e79bd66aa21f344390753dc's message to understand original and new approaches.

### Python

- ✅ Original renaming approach still works.
- ✅ New naming approach works.

### Rust

- ✅ Original renaming approach still works.
- ✅ New naming approach works.

**Comment by adamgreig** [comment]
This looks good to me. I propose we merge this first, then fix conflicts in #89 and merge that too, then release a new svdtools (despite the stm32-rs and rp2040 tests failing at that point) and get PRs merged in those repos as soon as possible afterwards to use the new version. Does that sound OK?

**Comment by burrbull** [comment]
OK

**Pull request merged**